dss_datamover/client_application.py

```diff
@@ -405,6 +406,21 @@ def message_server_index(self):
             except Exception as e:
                 self.logger.excep("Monitor-Index - {}".format(e))
 
+        while len(nfs_mount_retry_list) > 0:  # Retry failed NFS mounts for a period of time. 
+            retry_message, start_time = nfs_mount_retry_list.pop(0)
+            if (datetime.now() - start_time).seconds > NFS_MOUNT_FAIL_WAIT_TIME:
+                self.logger.error('Retry mounting {} failed in last {} mins, abort.'.format(retry_message["nfs_share"], NFS_MOUNT_FAIL_WAIT_TIME / 60))
+                self.mount_failed.value = 1
+            else:
+                if self.nfs_mount(retry_message["nfs_cluster"], retry_message["nfs_share"]):
+                    self.add_task(retry_message)
+                    is_index_data_added = True
+                    if retry_message["dir"] in self.index_buffer:
+                        self.index_buffer[retry_message["dir"]] += len(retry_message["files"])
+                    else:
+                        self.index_buffer[retry_message["dir"]] = len(retry_message["files"])
+                else:
+                    nfs_mount_retry_list.append((retry_message, start_time))
```

Comment:
could this lead to a potential infinite loop? Do we want to retry for a second or multiple time if nfs mount fails?

Reply:
I don't think this will lead to inifinite loop. The logic here is putting back the nfs message that failed in a retry, next time it will be poped out from the list again and it will be excluded if it reaches retry deadline. So it is guaranteed that after a certain amount of time, there will be no element left in the list.
Currently it will retry for a certain period of time.